cmd/stack_create_test.go

```diff
@@ -23,291 +22,129 @@ import (
-
-	args := []string{"stack", "create", testStackName, "--copy", "experimental/nodejs"}
-	_, err := cmdtest.RunAppsody(sandbox, args...)
-
-	if err == nil {
-		t.Fatal(err)
-	}
-
-	exists, err := cmdtest.Exists(testStackName)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if exists {
-		// It SHOULDN'T exist, but it might
-		err = os.RemoveAll(testStackName)
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Fatal(err)
-	}
-}
-
-func TestStackCreateInvalidStackCase4(t *testing.T) {
-	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
-	defer cleanup()
-
-	testStackName := "testing-stack-create-invalid-4"
-
-	args := []string{"stack", "create", testStackName, "--copy", "exp/java-microprofile"}
-	_, err := cmdtest.RunAppsody(sandbox, args...)
-
-	if err == nil {
-		t.Fatal(err)
-	}
-
-	exists, err := cmdtest.Exists(testStackName)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if exists {
-		// It SHOULDN'T exist, but it might
-		err = os.RemoveAll(testStackName)
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Fatal(err)
-	}
-}
-
-func TestStackCreateInvalidStackName(t *testing.T) {
-	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
-	defer cleanup()
-
-	testStackName := "testing_stack_invalid_name"
-
-	args := []string{"stack", "create", testStackName}
-	_, err := cmdtest.RunAppsody(sandbox, args...)
-
-	if err == nil {
-		t.Fatal(err)
-	}
-
-	exists, err := cmdtest.Exists(testStackName)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if exists {
-		// It SHOULDN'T exist, but it might
-		err = os.RemoveAll(testStackName)
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Fatal(err)
+func TestStackCreateValidCases(t *testing.T) {
+	var stackCreateValidTests = []struct {
+		testName string
+		args     []string // input
+	}{
+		{"No args", []string{}},
+		{"Existing default repo", []string{"--copy", "incubator/nodejs"}},
+	}
+	for _, testData := range stackCreateValidTests {
+		// need to set testData to a new variable scoped under the for loop
+		// otherwise tests run in parallel may get the wrong testData
+		// because the for loop reassigns it before the func runs
+		tt := testData
+		// call t.Run so that we can name and report on individual tests
+		t.Run(tt.testName, func(t *testing.T) {
+			sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
+			defer cleanup()
+			testStackName := "testing-create-stack"
+			args := []string{"stack", "create", testStackName, "--config", filepath.Join(sandbox.TestDataPath, "default_repository_config", "config.yaml")}
+			args = append(args, tt.args...)
+			_, err := cmdtest.RunAppsody(sandbox, args...)
+			if err != nil {
+				t.Fatal("Unexpected error when running appsody stack create: ", err)
+			}
+			exists, err := cmdtest.Exists(testStackName)
+			if err != nil {
+				t.Fatal("Failed to check if the stack exists: ", err)
+			}
+			if !exists {
+				t.Fatal("Stack doesn't exist despite appsody stack create executing correctly")
+			}
+			os.RemoveAll(testStackName)
+			if err != nil {
```

Comment:
This `err` is from the Exists command, does the `RemoveAll` return an error?

Reply:
Actually do we even have to remove? Since we are sandboxing each test?

Reply:
Good question! I guess not, actually? I can give it a go with this bit removed. Didn't really think about that when I took the functionality from each test to turn it into tables.

Reply:
It left a bunch of files in the `testdata` folder, which then failed the tests the second time running. I can look into why another time?

Reply:
Sounds good, but then would need to do `err = os.RemoveAll(testStackName)`

Reply:
Yes, of course!